internal/config/profile.go

```diff
@@ -175,6 +180,51 @@ func Exists(name string) bool {
 	return search.StringInSlice(List(), name)
 }
 
+// getConfigHostnameFromEnvs patches the agent hostname based on set env vars.
+func getConfigHostnameFromEnvs() string {
+	var builder strings.Builder
+
+	envVars := []struct {
+		envName  string
+		hostName string
+	}{
+		{AtlasActionHostNameEnv, AtlasActionHostName},
+		{GitHubActionsHostNameEnv, GitHubActionsHostName},
+		{ContainerizedHostNameEnv, DockerContainerHostName},
+	}
+
+	for _, envVar := range envVars {
+		if envIsPopulated(envVar.envName) {
+			appendToHostName(&builder, envVar.hostName)
+		} else {
+			appendToHostName(&builder, "-")
+		}
+	}
+	configHostName := builder.String()
+
+	if isDefaultHostName(configHostName) {
+		return NativeHostName
+	}
+	return configHostName
+}
+
+func envIsPopulated(env string) bool {
+	return os.Getenv(env) == "true"
+}
```

Comment:
sometimes people will use 1|0 so 
```suggestion
func envIsTrue(env string) bool {
	return IsTrue(os.Getenv(env))
}
```

Reply:
Thanks for the tip! Added